lib/PoolClusterFarm.js

```diff
@@ -0,0 +1,120 @@
+var Pool          = require('./Pool');
+var PoolConfig    = require('./PoolConfig');
+var PoolNamespace = require('./PoolNamespace');
+var PoolSelector  = require('./PoolSelector');
+var Util          = require('util');
+var EventEmitter  = require('events').EventEmitter;
+
+/**
+ * PoolClusterFarm
+ * @constructor
+ * @param {object} [config] The pool cluster farm configuration
+ * @public
+ */
+function PoolClusterFarm(config) {
+  EventEmitter.call(this);
+
+  config = config || {};
+  this._servers = [];
+  this._numMasters = 0;
+  this._numSlaves = 0;
+  this._last_index = 0;
+  this.debug = config.debug || false
+}
+
+Util.inherits(PoolClusterFarm,EventEmitter);
+
+PoolClusterFarm.prototype.query = function query(sql,values,cb) {
+	var self = this
+
+	if (!this._numMasters || this._numMasters < 1) {
+		if (cb && typeof cb == 'function') { return cb('No servers assigned to PoolClusterFarm. Please verify...'); }
+		else { return values('No servers assigned to PoolClusterFarm. Please verify...'); }
+	}
+
+	if (typeof sql != 'string' || sql.length < 7) {
+		if (cb && typeof cb == 'function') { return cb('Invalid sql statement. Please verify...'); }
+		else { return values('Invalid sql statement. Please verify...'); }
+	}
+
+	if (sql.substr(0,6).toLowerCase() == 'select') {
```

Comment:
Haha, I guess I wouldn't be allowed to use this module then ;) I'm not sure if there are really any users who sit down and flush out a giant code base of hand written SQL mixed in their JavaScript, though. I imagine that there are really only two main type of production callers (1) those who use an ORM system that generates SQL, but the same ORM already understands reads/writes and is better off using the pool cluster for better control and (2) those who write stored procedures and call those from applications (which works really well when you have different programming languages interacting with the same database), which also is better off using pool cluster.